compiler/mio-tflite260/src/Helper.cpp

```diff
@@ -23,22 +24,52 @@ namespace mio
 namespace tflite
 {
 
-// This will provide v3/v3a format neutral BuiltinOperator
+/**
+ * This will provide v3/v3a format neutral BuiltinOperator
+ *
+ * This function referenced
+ * https://github.com/tensorflow/tensorflow/blob/7d12007d7800d3714a02e05059f3ea602d1aec78/tensorflow/lite/schema/schema_utils.cc
+ * and
+ * https://github.com/tensorflow/tensorflow/commit/cf28969fb2ea72c9738cd3acdddc0b69fd7dff5e
+ */
 ::tflite::BuiltinOperator builtin_code_neutral(const ::tflite::OperatorCode *opcode)
 {
   assert(opcode != nullptr);
-  int8_t dp_code = opcode->deprecated_builtin_code();
-  // 127 is max of int8_t which is upper bound of v3 builtin_code
-  // NOTE TensorFlow uses 'BuiltinOperator_PLACEHOLDER_FOR_GREATER_OP_CODES' for 127
-  if (dp_code < 127 && dp_code >= 0)
-    return ::tflite::BuiltinOperator(dp_code);
-  return opcode->builtin_code();
+  return std::max(opcode->builtin_code(),
+                  static_cast<::tflite::BuiltinOperator>(opcode->deprecated_builtin_code()));
 }
 
+/**
+ * This function does not validate the relation between deprecated_builtin_code() and
+ * builtin_code() because information is not
+ * i.e., which value is stored when builtin_code() is larger than
+ * BuiltinOperator_PLACEHOLDER_FOR_GREATER_OP_CODES?
+ *
+ * According to
+ * https://github.com/tensorflow/tflite-micro/blob/25da8245417b834ced53df722d6a0f517eb425b4/tensorflow/lite/micro/test_helpers.cc#L154,
+ * 0 is stored.
+ *
+ * According to
+ * https://github.com/tensorflow/tensorflow/blob/1ab788fa8d08430be239ab970980b891ad7af494/tensorflow/lite/schema/schema_conversion_utils.cc#L46-L50,
+ * 127 is stored.
```

Comment:
```cxx
  int8_t deprecated_builtin_code =
      static_cast<int8_t>(BuiltinOperator_PLACEHOLDER_FOR_GREATER_OP_CODES);
  if (builtin_code < BuiltinOperator_PLACEHOLDER_FOR_GREATER_OP_CODES) {
    deprecated_builtin_code = static_cast<int8_t>(builtin_code);
  }
```
The code is like this from your URL.

(1) Adding a URL helps understanding, but need to jump to the code that is not convenient
(2) I think it's not 127 but `deprecated_builtin_code` set to `builtin_code` if it's below or same as 127.

The second URL (this) is from a conversion tool...
If the values are different, I'll follow the main code not this tool.
I would assume there is no big deal in difference or someday this tool may be revised from some problem.

Reply:
> (2) I think it's not 127 but deprecated_builtin_code set to builtin_code if it's below or same as 127.

I only focused on "opcode which is larger than 127" cases :D
For "opcode which is smaller than 127" cases, you are correct :)

That's why I wrote "i.e., which value is stored when builtin_code() is larger than BuiltinOperator_PLACEHOLDER_FOR_GREATER_OP_CODES?" in comment

---
>  someday this tool may be revised from some problem.

If we only care about [Second URL](https://github.com/tensorflow/tensorflow/blob/1ab788fa8d08430be239ab970980b891ad7af494/tensorflow/lite/schema/schema_conversion_utils.cc#L46-L50), I agree that there is no problem for now.
Then shall we go as-is? :D

Reply:
> Then shall we go as-is? :D

Um? didn't quite catch this. Cabn you update this PR (the comment) so that it is clear?

Reply:
Ah, I mean "if you don't want to change the implementation now, I will not change" :)

Reply:
I think `builtin_code_neutral` should be changed as in this PR as current implementation seems to have a problem.

I think your `is_valid` code is OK.
Existing one depends on `builtin_code_neutral` which current PR does't and looks more clear.

About the comment, I think it would be better to make simple and clear. (I assumed they were added for the reviewers not for landing... but just a guess)

Reply:
> About the comment, I think it would be better to make simple and clear. (I assumed they were added for the reviewers not for landing... but just a guess)

Half was for review, half was for landing.
But I think it would be better to remove the comments because it looks like... Too Much Information =)
Instead, I will leave reference links to #8239!